docs/src/main/asciidoc/extension-maturity-model.adoc

```diff
@@ -0,0 +1,89 @@
+
+[id="extension-maturity-model"]
+= A maturity model for Quarkus extensions
+include::_attributes.adoc[]
+:diataxis-type: concept
+:categories: writing-extensions
+:topics: extensions
+:summary: Quarkus extensions can do a lot, or a little. This guide explains some of the capabilities extension authors may wish to include.
+////
+The document header ends at the first blank line. Do not remove the blank line between the header and the abstract summary.
+////
+
+What makes a good Quarkus extension? What capabilities is a Quarkus extension expected to provide?
+
+* Runtime actually works
+** Works in JVM mode
+** Works in dev mode
+** Works in native
+* Configuration support
+* Dev service (if there is an external service dependency)
+* Dev UI
+* Use build-time application knowledge to eliminate wasteful runtime code paths (supersonic subatomic performance)
+* Use build-time application knowledge to remove boilerplate
+* Codestart application template
+
+The order in this model isn't exact. Different developers will have different views on what capabilities are most important. You may wish to (say) prioritise performance over enhancing your extension's Dev UI tile. That's fine!
+Also, not every step will apply to every extension. For example, you don't need a dev service if your extension doesn't depend on external services.
```

Comment:
I think it needs to be clear that you don't need to implement all of this before releasing your extension. It's more or less mentioned but people have a tendency to go over the top sometimes when developing in the open and we need to clarify it's OK to publish a first version that doesn't handle everything.

Reply:
Perfect! (sorry forgot about it!)